roles/openshift_hosted_facts/tasks/main.yml

```diff
@@ -0,0 +1,9 @@
+---
+- name: Set hosted facts
+  openshift_facts:
+    role: hosted
+    openshift_env: "{{ item | oo_openshift_env }}"
+  no_log: true
+  with_items:
+  - "{{ hostvars[inventory_hostname] }}"
+  - "{{ hostvars }}"
```

Comment:
Noting that I updated this to line up with how we collect cloudprovider facts. Top level hostvars must be checked for anyone consuming the roles directly and passing non-inventory vars.